src/test/groovy/com/synacy/gradprogram/spock/exercise/ShoppingServiceSpec.groovy

```diff
@@ -0,0 +1,69 @@
+package com.synacy.gradprogram.spock.exercise
+
+import com.synacy.gradprogram.spock.demo.User
+import spock.lang.Specification
+
+class ShoppingServiceSpec extends Specification {
+
+    ShoppingService shoppingService
+    Cart cart
+    User user
+    Order order
+    OrderingService orderingService
+    DeliveryService deliveryService
+    OrderRepository orderRepository
+    DeliveryRequestRepository deliveryRequestRepository
+
+    void setup() {
+        cart = Mock(Cart)
+        user = Mock(User)
+        order = Mock(Order)
+
+        orderRepository = Mock(OrderRepository)
+        deliveryRequestRepository = Mock(DeliveryRequestRepository)
+
+        orderingService = Mock(OrderingService)
+        deliveryService = Mock(DeliveryService)
+        shoppingService = new ShoppingService(orderingService, deliveryService,
+                orderRepository, deliveryRequestRepository)
+    }
+
+    def "BuyNonSpoilingItemsInCart should check if discount applies to cart and save a delivery request"() {
+        given:
+        user.getFirstName() >> "John"
+        user.getLastName() >> "Doe"
+        user.getAddress() >> "Philippines"
+
+        orderingService.createAnOrder(_,_,_,_) >> order
+
+        when:
+        shoppingService.buyNonSpoilingItemsInCart(cart, user)
+
+        then:
+        1 * orderingService.applyDiscountToCartItems(cart)
+
+        1 * deliveryService.createDelivery(order)
+    }
+
+    def "GetOrderSummary should return an order summary based on UUID"() {
+        given:
+        Order testOrder = new Order(totalCost: 10, status: OrderStatus.PENDING)
+        UUID testUuid = testOrder.getId()
+
+        DeliveryRequest testDeliveryRequest = new DeliveryRequest(orderId: testUuid,
+                deliveryDate: new Date(), courier: Courier.JRS)
```

Comment:
Let's try to favor using Mocks as much as possible, we can use Mocks on these two here.